mozetl/health_check/normandy/job.py

```diff
@@ -0,0 +1,204 @@
+            .crosstab("event_map_values[source]", "event_string_value")
+        )
+    )
+    runner_nested_dict = crosstab_to_dict(
+        events.select(events.event_map_values["source"], events.event_string_value)
+        .filter("event_map_values['source'] = 'normandy/runner'")
+        .crosstab("event_map_values[source]", "event_string_value")
+    )
+    assert len(runner_nested_dict) == 1
+    # only one runner, so discard "outer" dict
+    runner_health = runner_nested_dict["normandy/runner"]
+    status_totals = dict(
+        [
+            (row[0], row[1])
+            for row in events.groupBy(events.event_string_value)
+            .agg(F.count("*"))
+            .collect()
+        ]
+    )
+
+    def myzip(submission_date, value):
+        return zip(submission_date, value)
+
+    zipped_schema = T.ArrayType(
+        T.StructType(
+            [
+                T.StructField("submission_date", T.StringType()),
+                T.StructField("value", T.LongType()),
+            ]
+        )
+    )
+    myzip = F.udf(myzip, zipped_schema)
+
+    flat_scalar = (
+        longitudinal.select(
+            "submission_date", F.explode("scalar_parent_normandy_recipe_freshness")
+        )
+        .select("key", myzip("submission_date", "value").alias("time_value"))
+        .select("key", F.explode("time_value").alias("time_value"))
+    )
+    flat_scalar = flat_scalar.filter(
+        "time_value.submission_date >= '%s'" % start_longitudinal.strftime("%Y-%m-%d")
+    ).filter("time_value.value IS NOT NULL")
+    recipe_versions = nest_keys(
+        dict(
+            [
+                ((row[0], row[1]), row[2])
+                for row in flat_scalar.groupBy("key", "time_value.value")
+                .count()
+                .collect()
+            ]
+        )
+    )
+    return NormandyHealthCheckResult(
+        now, recipe_versions, recipe_health, action_health, runner_health, status_totals
+    )
+
+
+@click.command()
+@click.option("--period", default=6, help="length of the retention period in hours")
+@click.option(
+    "--slack", default=6, help="number of hours to account for submission latency"
+)
+def main(period, slack):
+    """Assess Normandy health for the given period.
+    """
+    spark = SparkSession.builder.appName("normandy-health-check").getOrCreate()
+    spark.conf.set("spark.sql.session.timeZone", "UTC")
+    s3 = boto3.client("s3")
+
+    events = spark.read.parquet("s3://telemetry-parquet/events/v1/")
+    now = arrow.get()
+    events_start = now.shift(hours=-slack)
+    longitudinal = spark.sql(
+        """
+        SELECT client_id, submission_date, scalar_parent_normandy_recipe_freshness
+        FROM longitudinal
+        """
+    )
+    longitudinal_start = events_start.shift(days=-1)
+    result = extract(events, events_start, longitudinal, longitudinal_start)
+    s3.put_object(Body=json.dump(result), Bucket=OUTPUT_BUCKET, Key=OUTPUT_KEY)
```

Comment:
I would make these parameters of the function when extracting it out, that way you can easily write test output to a test bucket / directory.